src/main/java/de/triology/cas/services/RegistryEtcd.java

```diff
@@ -77,6 +78,58 @@ public String getFqdn() {
         }
     }
 
+    public URI getCasLogoutUri(String doguname) throws GetCasLogoutUriException {
+        JSONObject doguMetaData, properties;
+        try {
+            EtcdKeysResponse.EtcdNode node = getDoguNodeFromEtcd(doguname);
+            doguMetaData = getCurrentDoguNode(node);
+            if (doguMetaData != null) {
+                properties = getPropertiesFromMetaData(doguMetaData);
+            } else {
+                throw new GetCasLogoutUriException("Could not get dogu metadata");
+            }
+            return getLogoutUriFromProperties(properties);
+        } catch (ParseException | URISyntaxException | GetDoguNodeFromEtcdException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1166)